src/Statistics.h

```diff
@@ -1,21 +1,169 @@
 
 // Statistics.h
 
-
-
-
 #pragma once
 
+// tolua_begin
+
+/*  Hello fellow developer !
+	In case you are trying to add new statistics to Cuberight you need to do a few things:
+	---------------------------------------------------------------------------
+	1. add a new entry to the enum class Statistic - as camelCase (first letter small)
+	2. add the value from the enum class to
+		- m_MainStats
+		- m_DirtyStatistics
+		-------------------- at the bottom of the .h file
+		- m_StatNames
+		- GetName-functions
+		-------------------- in the .cpp file
+		The String in m_StatNames and GetName is used for saving on disk!
+		so use the same string!
+
+	In case you want to add a mapping of network IDs to the used stats
+	you will find a lua script in ../Tools/BlockTypePaletteGenerator/ExportStatMapping.lua
+	it will provide you with information how to use it. you need a registries.json
+	exported from the server https://wiki.vg/Data_Generators
+
+
+	In Case Mojang changes the names exported by the server at any point
+	make sure only to change the names in enum class Statistic to keep the
+	old API intact and keep the order up to the point I noted
+
+		Greetings 12xx12 */
+
+enum class Statistic
+{
+	/*  Please do not change order till told it is allowed
+		this is used to map the old API to the new one */
 
+	/* Achievements */
+	achOpenInventory,     /* Taking Inventory     */
```

Comment:
Couldn't we leave this as the old `enum eStatistics`, and for each old value, do something like

```cpp
enum eStatistics
{
    achOpenInventory = Achievement::OpenInventory,
}
```

Reply:
This enum should also be in Registries/Statistics.h

Reply:
the old achievements ARE statistics and the "new" eStatistics is the old one just moved

gonna move them

Reply:
And tried this but CLion told me: 
"Integral constant expression must have integral or unscoped enumeration type, not 'Statistic'"

Reply:
hm. perhaps we need to throw the entire keyboard at the problem.

`achOpenInventory = static_cast<std::underlying_type_t<eStatistics>>(Statistic::OpenInventory)`

(though `achOpenInventory = static_cast<int>(Statistic::OpenInventory)` would probably work)

Reply:
at the moment the enum and the enum class are ordered the same way so you do simply do static_cast<enum class>(enum) to "convert"

Reply:
I see, I guess that could work then. My only concern is that if we autogenerate the list of new stats and the Lua script sorts them, we might miss any break in the ordering during code review

Reply:
the only thing i generate at the moment is the lookup statistic -> id to send. new statistics have to be added manually.

i wrote some instructions in the statistics.h how to do this ( and some instructions what not to do.

At some point I manually ordered put some numeric values to the entries then went back to the same order

and for the lookup the order doesn't matter - i sort them by the numeric return value. looks nicer

but I just removed I this from the API because it wasn't neiter documented - see latest comit

Reply:
and if we try to autogenerate all the lookups/enums/... we loose compability for old versions (removed statistics...)

Reply:
moved enum to registries/statistics.h